ui/text_displayer.js

```diff
@@ -204,6 +204,25 @@ shaka.ui.TextDisplayer = class {
     captionsStyle.color = cue.color;
     captionsStyle.direction = cue.direction;
 
+    if (cue.backgroundImage) {
+      captionsStyle.backgroundImage = 'url(\'' + cue.backgroundImage + '\')';
+      captionsStyle.backgroundRepeat = 'no-repeat';
+      captionsStyle.backgroundSize = 'contain';
+      captionsStyle.backgroundPosition = 'center';
+      if (cue.region) {
+        if (cue.region.heightUnits === shaka.text.CueRegion.units.PERCENTAGE) {
```

Comment:
To help communicate that the only difference between the branches is the units, I would suggest doing something like:

```
  const heightUnit = cue.region.heightUnits == shaka.text.CueRegion.units.PERCENTAGE ? '%' : 'px';
  const widthUnit = cue.region.widthUnits == shaka.text.CueRegion.units.PERCENTAGE ? '%' : 'px';

  captionsSytle.height = cue.region.height + heightUnit;
  captionsSytle.width = cue.region.width + widthUnit;
```